test/__mockData__/mockMjmlReactTestData.tsx

```diff
@@ -0,0 +1,360 @@
+          {"Second Line"}
+        </MjmlButton>
+      ),
+      expectedMjml: `<mj-button>First LineSecond Line</mj-button>`,
+    },
+  ],
+  MjmlDivider: [
+    {
+      mjmlReact: <MjmlDivider borderWidth={2} borderColor="red" />,
+      expectedMjml: `<mj-divider border-width="2px" border-color="red"></mj-divider>`,
+    },
+  ],
+  MjmlRaw: [
+    {
+      mjmlReact: (
+        <MjmlRaw>
+          <h1>Hello World!</h1>
+        </MjmlRaw>
+      ),
+      expectedMjml: `<mj-raw><h1>Hello World!</h1></mj-raw>`,
+    },
+  ],
+  MjmlImage: [
+    {
+      mjmlReact: <MjmlImage src="https://www.faire.com/logo.png" />,
+      expectedMjml: `<mj-image src="https://www.faire.com/logo.png"></mj-image>`,
+    },
+  ],
+  MjmlSpacer: [
+    {
+      mjmlReact: <MjmlSpacer height={10} />,
+      expectedMjml: `<mj-spacer height="10px"></mj-spacer>`,
+    },
+  ],
+  MjmlWrapper: [
+    {
+      mjmlReact: (
+        <MjmlWrapper padding={20} fullWidth>
+          <span>Hello World!</span>
+        </MjmlWrapper>
+      ),
+      expectedMjml: `<mj-wrapper padding="20px" full-width="full-width"><span>Hello World!</span></mj-wrapper>`,
+    },
+  ],
+  MjmlText: [
+    {
+      mjmlReact: (
+        <MjmlText fontWeight="20" letterSpacing={2}>
+          <span>Hello World!</span>
+        </MjmlText>
+      ),
+      expectedMjml: `<mj-text font-weight="20" letter-spacing="2px"><span>Hello World!</span></mj-text>`,
+    },
+  ],
+  MjmlHero: [
+    {
+      mjmlReact: (
+        <MjmlHero backgroundUrl="https://www.faire.com/logo.png">
+          <MjmlImage src="https://www.faire.com/logo.png" />
+        </MjmlHero>
+      ),
+      expectedMjml: `<mj-hero background-url="https://www.faire.com/logo.png"><mj-image src="https://www.faire.com/logo.png"></mj-image></mj-hero>`,
+    },
+  ],
+  MjmlClass: [
+    {
+      mjmlReact: <MjmlClass />,
+      expectedMjml: `<mj-class></mj-class>`,
+    },
+  ],
+  MjmlInclude: [
+    {
+      mjmlReact: <MjmlInclude />,
+      expectedMjml: `<mj-include></mj-include>`,
+    },
+  ],
+  MjmlHead: [
+    {
+      mjmlReact: <MjmlHead />,
+      expectedMjml: `<mj-head></mj-head>`,
+    },
+  ],
+  MjmlHtmlAttributes: [
+    {
+      mjmlReact: <MjmlHtmlAttributes />,
+      expectedMjml: `<mj-html-attributes></mj-html-attributes>`,
+    },
+  ],
+  MjmlFont: [
+    {
+      mjmlReact: (
+        <MjmlFont name="FaireFont" href="https://www.faire.com/faireFont" />
+      ),
+      expectedMjml: `<mj-font name="FaireFont" href="https://www.faire.com/faireFont"></mj-font>`,
+    },
+  ],
+  MjmlPreview: [
+    {
+      mjmlReact: <MjmlPreview>Hello World!</MjmlPreview>,
+      expectedMjml: `<mj-preview>Hello World!</mj-preview>`,
+    },
+  ],
+  MjmlStyle: [
+    {
+      mjmlReact: <MjmlStyle>{".button{}"}</MjmlStyle>,
+      expectedMjml: `<mj-style>.button{}</mj-style>`,
+    },
+    {
+      mjmlReact: <MjmlStyle>{"body > div {}"}</MjmlStyle>,
+      expectedMjml: `<mj-style>body &gt; div {}</mj-style>`,
```

Comment:
`&gt;` I think this is why we used the dangerouslySetInnerHtml for MjmlStyle. Probably not the best to silently do a dangerous operation but not sure how to handle this case.

I don't like the idea or requiring everyone to use `dangerouslySetInnerHTML` to use styles

Reply:
This is tricky. For this one either option is requiring everyone to use `dangerouslySetInnerHtml`, the one option just hides it under the hood.  I feel given `dangerouslySetInnerHtml` has the potential to be a security risk it is better to be super explicit that it is being used. But am open for more conversation around this